packages/location-management/src/components/LocationTree/index.tsx

```diff
@@ -1,26 +1,91 @@
-import React, { ChangeEvent, useState } from 'react';
-import { Input, Tree as AntTree } from 'antd';
+import React, { ChangeEvent, useEffect, useState } from 'react';
+import { Input, Tree as AntTree, Spin } from 'antd';
 import { SearchOutlined } from '@ant-design/icons';
 import reducerRegistry from '@onaio/redux-reducer-registry';
-import { ParsedHierarchyNode } from '../../ducks/types';
-import reducer, { reducerName } from '../../ducks/location-hierarchy';
-import { AntTreeProps } from '../LocationUnitView';
+import { ParsedHierarchyNode, RawOpenSRPHierarchy } from '../../ducks/types';
+import reducer, {
+  fetchAllHierarchies,
+  getAllHierarchiesArray,
+  reducerName,
+} from '../../ducks/location-hierarchy';
 import './tree.css';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchLocationUnits, LocationUnit } from '../../ducks/location-units';
+import { sendErrorNotification } from '@opensrp/notifications';
+import { generateJurisdictionTree } from './utils';
+import { LOCATION_HIERARCHY, LOCATION_UNIT_FINDBYPROPERTIES } from '../../constants';
+import { OpenSRPService } from '@opensrp/server-service';
 reducerRegistry.register(reducerName, reducer);
 
+const { getFilterParams } = OpenSRPService;
+
+export interface AntTreeProps {
+  title: JSX.Element;
+  key: string;
+  children: AntTreeProps[];
+}
+
 interface TreeProp {
-  data: ParsedHierarchyNode[];
+  opensrpBaseURL: string;
+  accessToken: string;
   OnItemClick: (item: ParsedHierarchyNode) => void;
 }
 
+/** Gets the hierarchy of the location units
+ *
+ * @param {Array<LocationUnit>} location - array of location units to get hierarchy of
+ * @param {string} accessToken - Access token to be used for requests
+ * @param {string} opensrpBaseURL - base url
+ * @returns {Promise<Array<RawOpenSRPHierarchy>>} array of RawOpenSRPHierarchy
```

Comment:
Should this be `Array<RawOpenSRPHierarchy` instead?

Reply:
its an asynchronous function so the returned will be a promise